code/node_modules/relationships_doc.js

```diff
@@ -7,22 +7,20 @@ try {
     throw new Error('DataSource codebase not found. This application is designed to nest within an existing X-DataSource repo.') 
 }
 
-const SOURCEDIR = '/code/docs/source'
-const RELFILE = `${SOURCEDIR}/Relationships.rst`
+const ABCDoc = require('abc_doc')
 
-class RelationshipsDoc {
 
-    constructor() {
-        this.doc = this.get_relationships_doc()
-        this.write().then(_ => {
-            this.check_written()
-        }).catch(err => {env.logger.debug(err)})
+class RelationshipsDoc extends ABCDoc {
+
+    constructor(source_directory) {
+        let target_file = `${source_directory}/relationships.rst`
+        super(source_directory, target_file)
```

Comment:
This line activates the constructor from the ABC, passing in the `target_file` var to the ABC constructor